### PR TITLE
Setting up accessibility linting

### PR DESCRIPTION
**Pull request by 20shivangi**
This PR made required changes to add required plugins `eslint-plugin-github` and `@github/prettier-config` for accessibility linting on required `.js` files under components and do required linting fixes.

As confirmed [here](https://github.slack.com/archives/C0FSWLQ0Y/p1675426375237999?thread_ts=1675171537.705219&cid=C0FSWLQ0Y), now there is no need of converting `.js` to `.jsx`, thereby this PR only add required plugins and address linting fixes.

Issue : https://github.com/github/npm/issues/7511

<br>

**You can review this PR by checking commits here : https://github.com/npm/documentation/pull/433/commits**

### CODE REVIEW

File: theme/.eslintrc.js [hunk 1]
```diff
@@ -16,6 +17,8 @@ module.exports = {
     'plugin:react/recommended',
     'plugin:jsx-a11y/recommended',
```
Comment (khiga8) [inline theme/.eslintrc.js:18]: Consider swapping this with [`plugin:github/react` ](plugin:github/react)[plugin:github/react](https://github.com/github/eslint-plugin-github/blob/main/lib/configs/react.js) which will pull in this plugin along with additional rules. This plugin comes with `eslint-plugin-github`.

Comment (20shivangi) [reply theme/.eslintrc.js]: Addressed

**Review by khiga8 — APPROVED** [review]
Great work 🚀

**Review by prabum3 — APPROVED** [review]

**Review by Seryozha95 — APPROVED** [review]

**Review by Arekn0870 — COMMENTED** [review]
50

**Review by Arekn0870 — COMMENTED** [review]
👌

**Pull request merged**